io/src/lib.rs

```diff
@@ -75,6 +110,15 @@ impl<R: std::io::Read> Read for R {
     }
 }
 
+#[cfg(feature = "std")]
+impl<R: std::io::BufRead + Read + ?Sized> BufRead for R {
```

Comment:
In b4696c449633e2078a79f71df3a9e75782afea89:

I see we already do this for `Read`, so I don't think we should hold up this PR for it, but does this code interact badly with the coherency rules? I think it means that users can implement this trait for their own types only if `std` is off ... and then if somebody turns `std` on, their code will break.

Reply:
I think the only way for it to happen is to depend on `std` and simultaneously on `bitcoin-io` without `std`. I think that shouldn't happen since activating `std` in a downstream crate ought to activate `std` here and in every other crate.

Reply:
Yeah, I think you're right.